packages/frontend/component/src/lit-react/lit-portal/lit-portal.tsx

```diff
@@ -98,47 +98,28 @@ export const useLitPortalFactory = () => {
             ? elementOrFactory()
             : elementOrFactory;
         return createLitPortalAnchor(event => {
-          const { name, target } = event;
-          const id = target.portalId;
-
-          if (name === 'connectedCallback') {
-            setPortals(portals => [
-              ...portals,
-              {
-                id,
-                portal: ReactDOM.createPortal(element, target, id),
-              },
-            ]);
-            return;
-          }
-
-          if (name === 'disconnectedCallback') {
-            setPortals(portals => portals.filter(p => p.id !== id));
-            return;
-          }
-
-          const prevId = event.previousPortalId;
-
-          // Ignores first `willUpdate`
-          if (!prevId) {
-            return;
-          }
-
           setPortals(portals => {
-            const portal = portals.find(p => p.id === prevId);
-            if (!portal) return [...portals];
-
-            // Updates `ID`
-            portal.id = id;
-
-            // Re-rendering
-            //  true: `inline link`
-            //  false: `pdf embed view`
-            if (rerendering) {
-              portal.portal = ReactDOM.createPortal(element, target, id);
+            const { name, target, previousPortalId } = event;
+            const id = target.portalId;
+
+            // skips update when
+            // - rerendering === false (for pdf preview etc)
+            // - previousPortalId is undefined (for the first willUpdate event)
+            if (name === 'willUpdate' && (!previousPortalId || !rerendering)) {
+              return portals;
             }
 
-            return [...portals];
+            const newPortals = portals.filter(
+              p => p.id !== previousPortalId && p.id !== id
```

Comment:
`LitReactPortal.portalId`  is always updated, so it cannot be deleted correctly when `disconnectedCallback` is used

Reply:
try debug here:

```ts
const newPortals = portals.filter(
    p => p.id !== previousPortalId && p.id !== id
);
console.log(portals.length, newPortals.length)
```

https://github.com/user-attachments/assets/8252467b-d2dd-4dda-ba97-dd51eac9f49f

Reply:
I see. Thanks!